pegen/pegen.c

```diff
@@ -592,3 +592,90 @@ alias_for_star(Parser *p)
     }
     return alias(str, NULL, p->arena);
 }
+
+void *
+seq_get_tail(void *previous, asdl_seq *seq)
+{
+    if (asdl_seq_LEN(seq) == 0) {
+        return previous;
+    }
+    return asdl_seq_GET(seq, asdl_seq_LEN(seq) - 1);
+}
+
+PegenAlias *
+pegen_alias(alias_ty alias, int line, int endline, int col, int endcol, PyArena *arena)
+{
+    PegenAlias *a = PyArena_Malloc(arena, sizeof(PegenAlias));
+    if (!a) {
+        return NULL;
+    }
+    a->alias = alias;
+    a->line = line;
+    a->endline = endline;
+    a->col = col;
+    a->endcol = col;
+    return a;
+}
+
+asdl_seq *seq_map_to_alias(Parser *p, asdl_seq *seq)
+{
+    int len = asdl_seq_LEN(seq);
+    asdl_seq *new_seq = _Py_asdl_seq_new(len, p->arena);
+    for (int i = 0; i < len; i++) {
+        PegenAlias *a = asdl_seq_GET(seq, i);
+        asdl_seq_SET(new_seq, i, a->alias);
+    }
+    return new_seq;
+}
+
+int
+expr_type(void *a, int head, int line)
```

Comment:
Why not change the type of `a` to `expr_ty`? If the caller passes a `void *` it'll still pass the compiler, and if something else is passed, it's probably an error. Then you won't need the casts in the function body. (Ditto below.)